### PR TITLE
tproxy remove cleanup controller

### PR DESCRIPTION
**Pull request by kschoche**
Changes proposed in this PR:
- remove cleanup controller code. 
- remove all flags related to cleanup controller.
- use `mgr.Start(ctx)` to run the webhook with a context and remove signal handling from connect command, and associated tests.
- adds an additional test to the `endpoints_controller_test` to provide coverage for creation of health checks when pods are in PodPending phase.

How I've tested this PR:
Manually deploy with an application and force-kill pods.
Unit tests pass.
Run connect acceptance tests against this build.

How I expect reviewers to test this PR:
Code review
Test by manually deploying and force/killing pods

Note to reviewers: I'm not 150% certain that test coverage couldn't be improved. We have existing test coverage for the deregister path, and it works with acceptance tests. Suggestions?

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

File: connect-inject/endpoints_controller_test.go [hunk 1]
```diff
@@ -446,6 +446,75 @@ func TestReconcileCreateEndpoint(t *testing.T) {
 			expectedProxySvcInstances:  []*api.CatalogService{},
 			expectedAgentHealthChecks:  nil,
 		},
+		{
+			name:          "PodPending endpoints",
```
Comment (kschoche) [inline connect-inject/endpoints_controller_test.go:450]:
```suggestion
			name:          "PodPending pods in endpoints",
```

**Review by thisisnotashwin — APPROVED** [review]
WOOT!! so good!!

File: connect-inject/endpoints_controller_test.go [hunk 2]
```diff
@@ -446,6 +446,75 @@ func TestReconcileCreateEndpoint(t *testing.T) {
...
+				pod1 := createPod("pod1", "1.2.3.4", true)
+				pod1.Status.Phase = corev1.PodPending
+				endpoint := &corev1.Endpoints{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "service-created",
+						Namespace: "default",
+					},
+					Subsets: []corev1.EndpointSubset{
+						corev1.EndpointSubset{
+							Addresses: []corev1.EndpointAddress{
```
[the remark this answers is not in the document]
Comment (thisisnotashwin) [reply connect-inject/endpoints_controller_test.go:462]: haha!! that makes things even more convenient. Delete more code from the PR 😛

[the remark this answers is not in the document]
Comment (kschoche) [reply connect-inject/endpoints_controller_test.go:462]: Oooo! Great catch Iryna! I will remove that test and also remove the code in the endpoint controller which handles that specific case. Thanks!

[the remark this answers is not in the document]
Comment (kschoche) [reply connect-inject/endpoints_controller_test.go]: On second thought, I've removed the test so this PR is only about the cleanup controller and will submit a 2nd PR to remove the PodPending handling (which is a no-op currently) from the endpoints controller!

**Pull request merged**